go/vt/vtgate/vtgateconntest/client.go

```diff
@@ -698,6 +698,35 @@ func (f *fakeVTGateService) ResolveTransaction(ctx context.Context, dtid string)
 	return nil
 }
 
+func (f *fakeVTGateService) MessageStream(ctx context.Context, keyspace string, shard string, keyRange *topodatapb.KeyRange, name string, sendReply func(*sqltypes.Result) error) error {
+	if f.hasError {
+		return errTestVtGateError
+	}
+	if f.panics {
+		panic(fmt.Errorf("test forced panic"))
+	}
+	f.checkCallerID(ctx, "ResolveTransaction")
+	if name != messageName {
+		return errors.New("MessageStream name mismatch")
+	}
+	sendReply(messageStreamResult)
+	return nil
+}
+
+func (f *fakeVTGateService) MessageAck(ctx context.Context, keyspace string, name string, ids []*querypb.Value) (int64, error) {
+	if f.hasError {
+		return 0, errTestVtGateError
+	}
+	if f.panics {
+		panic(fmt.Errorf("test forced panic"))
+	}
+	f.checkCallerID(ctx, "ResolveTransaction")
+	if !reflect.DeepEqual(ids, messageids) {
+		return 0, errors.New("MessageAck ids mismatch")
+	}
+	return 1, nil
```

Comment:
Make this 1 a constant below, and test for it at the other spot where I have the other comment.